packages/mermaid/src/diagrams/xychart/chartBuilder/components/axis/BaseAxis.ts

```diff
@@ -0,0 +1,342 @@
+import { XYChartAxisConfig } from '../../../../../config.type.js';
+import { log } from '../../../../../logger.js';
+import {
+  BoundingRect,
+  Dimension,
+  DrawableElem,
+  Point,
+  XYChartAxisThemeConfig,
+} from '../../Interfaces.js';
+import { ITextDimensionCalculator } from '../../TextDimensionCalculator.js';
+import { AxisPosition, IAxis } from './index.js';
+
+export abstract class BaseAxis implements IAxis {
+  protected boundingRect: BoundingRect = { x: 0, y: 0, width: 0, height: 0 };
+  protected axisPosition: AxisPosition = 'left';
+  private range: [number, number];
+  protected showTitle = false;
+  protected showLabel = false;
+  protected showTick = false;
+  protected outerPadding = 0;
+
+  constructor(
+    protected axisConfig: XYChartAxisConfig,
+    protected title: string,
+    protected textDimensionCalculator: ITextDimensionCalculator,
+    protected axisThemeConfig: XYChartAxisThemeConfig
+  ) {
+    this.range = [0, 10];
+    this.boundingRect = { x: 0, y: 0, width: 0, height: 0 };
+    this.axisPosition = 'left';
+  }
+
+  setRange(range: [number, number]): void {
+    this.range = range;
+    this.recalculateScale();
+  }
+
+  getRange(): [number, number] {
+    return [this.range[0] + this.outerPadding, this.range[1] - this.outerPadding];
+  }
+
+  setAxisPosition(axisPosition: AxisPosition): void {
+    this.axisPosition = axisPosition;
+  }
+
+  abstract getScaleValue(value: number | string): number;
+
+  abstract recalculateScale(): void;
+
+  abstract getTickValues(): Array<string | number>;
+
+  getTickDistance(): number {
+    return Math.abs(this.range[0] - this.range[1]) / this.getTickValues().length;
+  }
+
+  getAxisOuterPadding(): number {
+    return this.outerPadding;
+  }
+
+  private getLabelDimension(): Dimension {
+    return this.textDimensionCalculator.getDimension(
+      this.getTickValues().map((tick) => tick.toString()),
+      this.axisConfig.labelFontSize
+    );
+  }
+
+  recalculateOuterPaddingToDrawBar(): void {
+    if (0.7 * this.getTickDistance() > this.outerPadding * 2) {
```

Comment:
Move 0.7 to a constant to avoid magic numbers.

Reply:
Done

Reply:
Fixed